heareval/tasks/pipeline.py

```diff
@@ -556,6 +557,119 @@ def run(self):
+    Requires:
+        subsample_splits (list(SubsampleSplit)): task subsamples each split
+    """
+
+    sample_rates = luigi.Parameter()
+    metadata = luigi.TaskParameter()
+
+    def requires(self):
+        # Perform resampling on each split and sampling rate independently
+        resample_splits = [
+            ResampleSubCorpus(
+                sr=sr,
+                split=split["name"],
+                metadata=self.metadata,
+                data_config=self.data_config,
+            )
+            for sr in self.sample_rates
+            for split in self.data_config["splits"]
+        ]
+        return resample_splits
+
+    def run(self):
+        workdir = Path(self.workdir)
+        workdir.rmdir()
+        # We need to link the workdir of the requires, they will all be the same
+        # for all the requires so just grab the first one.
+        requires_workdir = Path(self.requires()[0].workdir).absolute()
+        workdir.symlink_to(requires_workdir)
+        self.mark_complete()
+
+
+class GenerateAudioStats(WorkTask):
+    """
+    Generates the statistics of the extracted and sampled audio files
+    Parameters
+        metadata (ExtractMetadata): task which extracts a corpus level metadata
+    Requires
+        traintestmeta(SplitTrainTestMetadata): task which produces the split
+            level metadata
+    """
+
+    sample_rates = luigi.Parameter()
+    metadata = luigi.TaskParameter()
+
+    def requires(self):
+        return {
+            "resample": ResampleSubcorpuses(
+                sample_rates=self.sample_rates,
+                metadata=self.metadata,
+                data_config=self.data_config,
+            ),
+            "metadata": self.metadata,
+        }
+
+    def get_metadata(self):
+        metadata = pd.read_csv(
+            self.requires()["metadata"].workdir.joinpath(
+                self.requires()["metadata"].outfile
+            )
+        )
+        return metadata
+
+    @staticmethod
+    def get_stats(audiofiles):
+        """Product the summary of a list of audio directory as dict"""
+        audio_stats = list(map(audio_util.get_audiostats, audiofiles))
+        durations = [stats["duration"] for stats in audio_stats]
+        sample_rates = [stats["sample_rate"] for stats in audio_stats]
+
+        stats = {
+            "total_audiofiles": len(durations),
+            "mean_samplerate": np.mean(sample_rates),
+            "mean_dur(sec)": np.mean(durations),
+            "median_dur)sec)": np.median(durations),
+        }
+        stats.update(
+            {
+                f"{str(p)}th percentile dur(sec)": np.percentile(durations, p)
+                for p in [10, 25, 75, 90]
+            }
+        )
+        return stats
+
+    def run(self):
+        master_stats = {"extracted": {}, "sampled": {}}
+        master_stats["extracted"] = self.get_stats(
+            self.get_metadata()["relpath"].apply(Path).tolist()
+        )
```

Comment:
There are two parts to the stats 
- `extracted`: This is the summary of the list of `relpath` of audio files in the metadata file which has not been processed at all. `relpath` is a reference to all the extracted files(used this way because it helpes avoid looking out for different `extracted workdir` subfolders in a different dataset. The metadata always contain all the files with correct relative path)
- `sampled`: This is a dict and contains the stats for all the partitions and the sampling rates.